pkg/stanza/fileconsumer/internal/reader/reader.go

```diff
@@ -106,9 +106,89 @@ func (r *Reader) ReadToEnd(ctx context.Context) {
 		}
 	}()
 
+	doneReadingFile := r.readHeader(ctx)
```

Comment:
Can we just check if the function needs to be called here? Seems like calling a function and producing a log is unnecessary overhead that every user has to pay for the tiny fraction of use cases which actually have a header.

Reply:
It's a debug log, but I'm fine removing it. Changed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35870/commits/7f56b47a5f1dafcc01f362aefcff0366ca852837. Please let me know if this is what you had in mind.